lib/ios/RNNSideMenu/MMDrawerController/MMDrawerController.m

```diff
@@ -323,52 +330,113 @@ - (void)closeDrawerAnimated:(BOOL)animated
         }
     } else {
         [self setAnimatingDrawer:animated];
-        CGRect newFrame = self.childControllerContainerView.bounds;
-
-        CGFloat distance = ABS(CGRectGetMinX(self.centerContainerView.frame));
-        NSTimeInterval duration = MAX(distance / ABS(velocity), MMDrawerMinimumAnimationDuration);
-
-        BOOL leftDrawerVisible = CGRectGetMinX(self.centerContainerView.frame) > 0;
-        BOOL rightDrawerVisible = CGRectGetMinX(self.centerContainerView.frame) < 0;
-
-        MMDrawerSide visibleSide = MMDrawerSideNone;
-        CGFloat percentVisble = 0.0;
-
-        if (leftDrawerVisible) {
-            CGFloat visibleDrawerPoints = CGRectGetMinX(self.centerContainerView.frame);
-            percentVisble = MAX(0.0, visibleDrawerPoints / self.maximumLeftDrawerWidth);
-            visibleSide = MMDrawerSideLeft;
-        } else if (rightDrawerVisible) {
-            CGFloat visibleDrawerPoints = CGRectGetWidth(self.centerContainerView.frame) -
-                                          CGRectGetMaxX(self.centerContainerView.frame);
-            percentVisble = MAX(0.0, visibleDrawerPoints / self.maximumRightDrawerWidth);
-            visibleSide = MMDrawerSideRight;
+        MMDrawerSide visibleSide = self.openSide;
+        
+        if (visibleSide == MMDrawerSideNone) {
+            [self setAnimatingDrawer:NO];
+            if (completion) {
+                completion(NO);
+            }
+            return;
         }
-
-        UIViewController *sideDrawerViewController =
-            [self sideDrawerViewControllerForSide:visibleSide];
-
-        [self updateDrawerVisualStateForDrawerSide:visibleSide percentVisible:percentVisble];
-
+        
+        UIViewController *sideDrawerViewController = [self sideDrawerViewControllerForSide:visibleSide];
         [sideDrawerViewController beginAppearanceTransition:NO animated:animated];
-
-        [UIView animateWithDuration:(animated ? duration : 0.0)
-            delay:0.0
-            options:options
-            animations:^{
-              [self setNeedsStatusBarAppearanceUpdateIfSupported];
-              [self.centerContainerView setFrame:newFrame withLayoutAlpha:0.0];
-              [self updateDrawerVisualStateForDrawerSide:visibleSide percentVisible:0.0];
+        
+        MMDrawerOpenMode openMode = (visibleSide == MMDrawerSideLeft) ? 
+                             self.leftDrawerOpenMode : 
+                             self.rightDrawerOpenMode;
+        
+        if (openMode == MMDrawerOpenModeAboveContent) {
+            // OVERLAY MODE
```

Comment:
2 separate functions for closing the overlay in each mode (maybe even just 1 closing logic func can be enough?...)